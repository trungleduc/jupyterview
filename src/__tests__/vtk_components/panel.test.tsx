import LeftPanel from "../../vtk_components/panel";
import { mount, shallow, render, ShallowWrapper } from "enzyme";

import React from "react";
import {mock_send_msg, MockModel} from "../../test_setup/utils"
describe("Test LeftPanel component", () => {
  let component: ShallowWrapper
  beforeAll(() => {
    const  model = new MockModel()
    component = shallow(<LeftPanel selectedId = {0} />);
  })

  it("should render correctly with mock props", () => {
    expect(component).toMatchSnapshot();
  });

  it("should have correct initial state", () => {
    expect(component.state("activePanelOnly")).toEqual(false)
    expect(component.state("animate")).toEqual(true)
    expect(component.state("navbarTabId")).toEqual("Home")
    expect(component.state("vertical")).toEqual(false)
    expect(component.state("selectedId")).toEqual("rx")
  });

});
