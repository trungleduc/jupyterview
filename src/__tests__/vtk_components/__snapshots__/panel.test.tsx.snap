// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LeftPanel component should render correctly with mock props 1`] = `
<Blueprint3.Tabs
  animate={true}
  className="left-panel"
  id="TabsExample"
  key="horizontal"
  large={false}
  onChange={[Function]}
  renderActiveTabPanelOnly={false}
  selectedTabId="rx"
  vertical={false}
>
  <Blueprint3.Tab
    className="left-panel-tab"
    disabled={false}
    id="rx"
    panel={<SourcePanel />}
    title="SOURCE"
  />
  <Blueprint3.Tab
    className="left-panel-tab"
    disabled={false}
    id="ng"
    panel={<ToolsPanel />}
    title="TOOLS"
  />
  <Blueprint3.Tab
    className="left-panel-tab"
    disabled={false}
    id="mb"
    panel={<InfoPanel />}
    title="INFO"
  />
</Blueprint3.Tabs>
`;
